lib/matplotlib/backends/qt_editor/figureoptions.py

```diff
@@ -236,12 +236,26 @@ def apply_callback(data):
         # re-generate legend, if checkbox is checked
         if generate_legend:
             draggable = None
-            ncol = 1
+            kwargs = {}
             if axes.legend_ is not None:
                 old_legend = axes.get_legend()
                 draggable = old_legend._draggable is not None
-                ncol = old_legend._ncol
-            new_legend = axes.legend(ncol=ncol)
+                kwargs = {name: getattr(old_legend, name) for name in
+                                ["numpoints", "markerscale", "shadow",
+                                 "columnspacing", "scatterpoints",
+                                 "handleheight", "borderpad",
+                                 "labelspacing", "handlelength",
+                                 "handletextpad", "borderaxespad"]}
+
+                kwargs['ncol'] = old_legend._ncol
+                kwargs['loc'] = old_legend._get_loc()
+                kwargs['mode'] = old_legend._mode
+                kwargs['title'] = old_legend.get_title().get_text()
+                kwargs['frameon'] = old_legend.get_frame_on()
+                kwargs['bbox_to_anchor'] = old_legend._bbox
```

Comment:
How does this work without `bbox_transform`?

Reply:
Hi @QuLogic, bbox_transform was used and then loc/bbox_to_anchor was updated. Since we directly update 'loc' and 'bbox_to_anchor' from the old legend, we don't need to do bbox_transform again. Please let me know if you have any questions, or some suggestions from your end.

Reply:
`bbox_to_anchor` and `bbox_transform` are combined to produce `Legend._bbox_to_anchor` though; why does this copy `Legend._bbox`? (Maybe this is stale and needs a rebase?)